label_studio/data_export/api.py

```diff
@@ -436,23 +442,109 @@ def get_queryset(self):
-        response['Content-Disposition'] = f'attachment; filename="{file_.name}"'
-        response['filename'] = file_.name
-        return response
+            ext = file_.name.split('.')[-1]
+
+            response = RangedFileResponse(request, file_, content_type=f'application/{ext}')
+            response['Content-Disposition'] = f'attachment; filename="{file_.name}"'
+            response['filename'] = file_.name
+            return response
+
+
+def async_convert(converted_format_id, export_type, **kwargs):
+    with transaction.atomic():
+        try:
+            converted_format = ConvertedFormat.objects.get(id=converted_format_id)
+        except ConvertedFormat.DoesNotExist:
+            logger.error(f'ConvertedFormat with id {converted_format_id} not found, conversion failed')
+            return
+        if converted_format.status != ConvertedFormat.Status.CREATED:
+            logger.error(f'Converson for export id {snapshot_id} to {export_type} already started')
+            return
+        converted_format.status = ConvertedFormat.Status.IN_PROGRESS
+        converted_format.save(update_fields=['status'])
+
+    snapshot = converted_format.export
+    project = snapshot.project
+    converted_file = snapshot.convert_file(export_type)
+    md5 = Export.eval_md5(converted_file)
+
+    now = datetime.now()
+    file_name = f'project-{project.id}-at-{now.strftime("%Y-%m-%d-%H-%M")}-{md5[0:8]}.{export_type.lower()}'
+    file_path = (
+        f'{project.id}/{file_name}'
+    )  # finally file will be in settings.DELAYED_EXPORT_DIR/project.id/file_name
+    file_ = File(converted_file, name=file_path)
+    converted_format.file.save(file_path, file_)
+    converted_format.status = ConvertedFormat.Status.COMPLETED
+    converted_format.save(update_fields=['file', 'status'])
+
+
+def set_convert_background_failure(job, connection, type, value, traceback):
+    from data_export.models import ConvertedFormat
+
+    convert_id = job.args[0]
+    ConvertedFormat.objects.filter(id=convert_id).update(status=Export.Status.FAILED)
+
+
+class ExportConvertAPI(generics.RetrieveAPIView):
+    queryset = Export.objects.all()
+    lookup_url_kwarg = 'export_pk'
+    permission_required = all_permissions.projects_change
+
+    def post(self, request, *args, **kwargs):
+        snapshot = self.get_object()
+        serializer = ExportConvertSerializer(data=request.data, context={'project': snapshot.project})
+        serializer.is_valid(raise_exception=True)
+        export_type = serializer.validated_data['export_type']
+
+        with transaction.atomic():
+            converted_format, created = ConvertedFormat.objects.get_or_create(
+                export=snapshot, export_type=export_type
+            )
+            if not created:
+                raise ValidationError(f'Converson to {export_type} already started')
+
+        start_job_async_or_sync(
+            async_convert,
+            converted_format.id,
+            export_type,
```

Comment:
add project here, it will be used for proper displaying of this job on django-rq page

Reply:
Fixed